test/archetype.test.ts

```diff
@@ -872,14 +876,256 @@ describe("Factory", function () {
     });
     await expect((await nft.connect(owner).config()).discounts).to.deep.equal(_discount);
     await nft.connect(owner).lockDiscounts("forever");
     await expect(nft.connect(owner).setDiscounts(discount)).to.be.reverted;
   });
+
+  it("test burn to mint functionality", async function () {
+    const [accountZero, accountOne] = await ethers.getSigners();
+
+    const owner = accountZero;
+    const minter = accountOne;
+
+    const newCollectionBurn = await factory.createCollection(
+      owner.address,
+      DEFAULT_NAME,
+      DEFAULT_SYMBOL,
+      DEFAULT_CONFIG
+    );
+    const resultBurn = await newCollectionBurn.wait();
+    const newCollectionAddressBurn = resultBurn.events[0].address || "";
+    const NFTBurn = await ethers.getContractFactory("Archetype");
+    const nftBurn = NFTBurn.attach(newCollectionAddressBurn);
+
+    const newCollectionMint = await factory.createCollection(
+      owner.address,
+      DEFAULT_NAME,
+      DEFAULT_SYMBOL,
+      DEFAULT_CONFIG
+    );
+    const resultMint = await newCollectionMint.wait();
+    const newCollectionAddressMint = resultMint.events[0].address || "";
+    const NFTMint = await ethers.getContractFactory("Archetype");
+    const nftMint = NFTMint.attach(newCollectionAddressMint);
+
+    await nftBurn.connect(owner).enableBurnToMint(nftMint.address, 2, 0, 5000);
+    await nftMint.connect(owner).setInvite(ethers.constants.HashZero, ipfsh.ctod(CID_ZERO), {
+      price: 0,
+      start: ethers.BigNumber.from(Math.floor(Date.now() / 1000)),
+      limit: 300,
+    });
+
+    // mint 10 tokens
+    await nftMint
+      .connect(minter)
+      .mint({ key: ethers.constants.HashZero, proof: [] }, 10, ZERO, "0x", {
+        value: 0,
+      });
+
+    // approve nftBurn to transfer tokens
+    await nftMint.connect(minter).setApprovalForAll(nftBurn.address, true);
+
+    // transfer away a token
+    await nftMint.connect(minter).transferFrom(minter.address, owner.address, 10);
+
+    // try to burn unowned token
+    await expect(nftBurn.connect(minter).burnToMint([9, 10])).to.be.revertedWith("NotTokenOwner");
+
+    // try to burn invalid number of tokens
+    await expect(nftBurn.connect(minter).burnToMint([9])).to.be.revertedWith(
+      "InvalidAmountOfTokens"
+    );
+
+    // burn 2 tokens and collect 1 token in new collection
+    await nftBurn.connect(minter).burnToMint([2, 4]);
+
+    // burn 4 tokens and collect 2 tokens in new collection
+    await nftBurn.connect(minter).burnToMint([1, 3, 5, 8]);
+
+    await expect(await nftMint.ownerOf(1)).to.be.equal(BURN);
+    await expect(await nftMint.ownerOf(2)).to.be.equal(BURN);
+    await expect(await nftMint.ownerOf(3)).to.be.equal(BURN);
+    await expect(await nftMint.ownerOf(4)).to.be.equal(BURN);
+    await expect(await nftMint.ownerOf(5)).to.be.equal(BURN);
+    await expect(await nftMint.ownerOf(8)).to.be.equal(BURN);
+    await expect(await nftMint.balanceOf(minter.address)).to.be.equal(3);
```

Comment:
don't we also need to check that the burner's nftBurn balance is 3. 

`nftBurn.balanceOf(minter.address).to.be.equal(3)`